Pegasus/include/Geometry.hpp

```diff
@@ -9,237 +9,251 @@
+    glm::dvec3 aVertex;
+    glm::dvec3 bVertex;
+    glm::dvec3 cVertex;
+    glm::dvec3 normal;
 
-    void SetAxes(glm::dvec3 const& a, glm::dvec3 const& b, glm::dvec3 const& c);
+    Triangle();
 
-    void GetAxes(glm::dvec3& a, glm::dvec3& b, glm::dvec3& c) const;
-
-    glm::dvec3 const& GetNormal() const;
-
-private:
-    glm::dvec3 m_aVertex;
-    glm::dvec3 m_bVertex;
-    glm::dvec3 m_cVertex;
-    glm::dvec3 m_normal;
+    Triangle(glm::dvec3 const& centerOfMass, glm::dvec3 const& a, glm::dvec3 const& b, glm::dvec3 const& c);
 
+    /** Calculates normal from member vertices and writes it to the normal member field */
     void CalculateNormal();
 };
 
+/** Sphere data storage class */
 class Sphere : public SimpleShape
 {
 public:
-    Sphere(glm::dvec3 const& centerOfMass, double r);
+    double radius;
 
-    void SetRadius(double r);
+    Sphere();
 
-    double GetRadius() const;
-
-private:
-    double m_radius;
+    Sphere(glm::dvec3 const& centerOfMass, double r);    
 };
 
+/** Cone data storage class */
 class Cone : public SimpleShape
 {
 public:
-    Cone(glm::dvec3 const& centerOfMass, glm::dvec3 const& a, double r);
-
-    void SetAppex(glm::dvec3 const& a);
-
-    glm::dvec3 const& GetAppex() const;
+    glm::dvec3 appex;
+    double radius;
 
-    void SetRadius(double r);
+    Cone();
 
-    double GetRadius() const;
-
-private:
-    glm::dvec3 m_appex;
-    double m_radius;
+    Cone(glm::dvec3 const& centerOfMass, glm::dvec3 const& a, double r);
 };
 
+/** Capsule data storage class */
 class Capsule : public SimpleShape
 {
 public:
-    Capsule(glm::dvec3 const& centerOfMass, glm::dvec3 const& halfHeight, double r);
-
-    void SetHalfHeight(glm::dvec3 const& halfHeight);
+    glm::dvec3 halfHeight;
+    double radius;
 
-    glm::dvec3 const& GetHalfHeight() const;
+    Capsule();
 
-    void SetRadius(double r);
-
-    double GetRadius() const;
-
-private:
-    glm::dvec3 m_halfHeight;
-    double m_radius;
+    Capsule(glm::dvec3 const& centerOfMass, glm::dvec3 const& halfHeight, double r);   
 };
 
-class Cylinder : public Capsule
+class Cylinder : public SimpleShape
 {
 public:
+    glm::dvec3 halfHeight;
+    double radius;
+
+    Cylinder();
+
     Cylinder(glm::dvec3 const& centerOfMass, glm::dvec3 const& halfHeight, double r);
 };
 
+/** Box data storage class */
 class Box : public SimpleShape
 {
 public:
-    Box(glm::dvec3 const& centerOfMass, glm::dvec3 const& a, glm::dvec3 const& b, glm::dvec3 const& c);
+    glm::dvec3 iAxis;
+    glm::dvec3 jAxis;
+    glm::dvec3 kAxis;
 
-    void SetAxes(glm::dvec3 const& a, glm::dvec3 const& b, glm::dvec3 const& c);
+    Box();
 
-    void GetAxes(glm::dvec3& a, glm::dvec3& b, glm::dvec3& c) const;
-
-private:
-    glm::dvec3 m_aAxis;
-    glm::dvec3 m_bAxis;
-    glm::dvec3 m_cAxis;
+    Box(glm::dvec3 const& centerOfMass, glm::dvec3 const& a, glm::dvec3 const& b, glm::dvec3 const& c);    
```

Comment:
how are `a`, `b` and `c` connected to `iAxis`, `jAxis` and `kAxis`? Should they be renamed to `i`, `j` and `k`?